marahgh/assignment1/Q2.cpp

```diff
@@ -0,0 +1,128 @@
+
+#include <iostream>
+#include <ext/slist>
+
+using namespace std;
+
+
+/*Node's list class */
+template <typename T>
+class Node{
+    T data;
+    Node* next;
+    public:
+    Node(T data):data(data),next(nullptr){
+    }
+    Node* getNext(){
+       return next; 
+    }
+    T& getData(){
+        return data;
+    }
+	void setData(T& data){
+		data=data;
+	}
+	void setNext(Node* node){
+		next=node;
+	}
+	
+};
+//----------------------------------------------------------------------------------//
+/* A generic list with basic functionality, constructor, destructor,
+    insertion to the end if the list and removing the last element of the list*/
+template< typename T>
+class List{
+    
+	Node<T>* head;
+	Node<T>* tail;
+	int size;
+	
+	public:
+	
+	List():head(nullptr),tail(nullptr),size(0){}
+	
+	Node<T>* getHead(){
+	    if(!this){
+	       return nullptr;
+	    }
+	    return head;
+	}
+	
+	int getSize(){
+	    if(!this){
+	        return -1;
+	    }
+        return size;
+    }
+	
+	void insertLast(T& data){
+	    Node<T>* nodeToInsert = new Node<T>(data);
+	    nodeToInsert->setNext(nullptr);
+		if(!head){
+			head=nodeToInsert;
+			tail=nodeToInsert;
+			size=1;
+			return;
+		}
+		tail->setNext(nodeToInsert);
+		tail=nodeToInsert;
+		size++;
+	}
+
+    void removeLast(){
+        if(!head){
+            return;
+        }
+        Node<T>* curr=head;
+        Node<T>* beforeLast=nullptr;
+        if(curr->getNext()==nullptr){
+            delete head;
+            head=nullptr;
+            tail=nullptr;
+            size=0;
+            return;
+        }
+        while (curr){
+            if(curr->getNext()== tail){
+                beforeLast=curr;

```

Comment:
NIT: You don't need the extra variable, you can use curr directly after the loop.